src/net.rs

```diff
@@ -438,78 +510,74 @@ pub trait NetworkQueue<T: Ord+Sequenced> {
     }
 
     fn clear(&mut self) {
-        self.as_queue_type_mut().clear()
+        self.as_queue_type_mut().clear();
     }
 
+    fn remove(&mut self, pkt: &T) -> Option<T>;
+
     fn discard_older_items(&mut self);
-    fn buffer_item(&mut self, item: T);
+    fn buffer_item(&mut self, item: T) -> bool;
     fn as_queue_type(&self) -> &ItemQueue<T>;
     fn as_queue_type_mut(&mut self) -> &mut ItemQueue<T>;
 }
 
-type ItemQueue<T> = VecDeque<T>;
-
-pub struct TXQueue {
-    pub queue: ItemQueue<Packet>,
-}
-
-pub struct RXQueue<T> {
-    pub queue: ItemQueue<T>,
-    pub buffer_wrap_index: Option<usize>
+pub struct Timestamp {
```

Comment:
Can we think of another name for this? A timestamp is just a point in time. This is something more.

Reply:
What about `NetAttempt`?